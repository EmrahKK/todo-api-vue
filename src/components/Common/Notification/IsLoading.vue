<template>
  <div class="loadingElement" v-bind:class="{loading: isLoading}">
    Loading...
  </div>
</template>

<script>
import { mapGetters } from 'vuex'

export default {
  computed: mapGetters([
    'isLoading'
  ])
}
</script>

<style scoped lang="less">
.loadingElement {
  position: fixed;
  bottom: -100%;
  left: 0;
  z-index: 90;
  width: 100%;
  font-size: 0.8em;
  background: #eaebf6;
  text-align: center;
  padding: 8pt 16pt;
  font-weight: bold;
  color: #7483a8;
  border-radius: 4pt;
  opacity: 0;
  transition: 1s;

  &.loading {
    display: block;
    bottom: 0;
    opacity: 1;
  }
}
</style>
