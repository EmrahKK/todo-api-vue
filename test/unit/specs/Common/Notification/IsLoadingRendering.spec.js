import Vue from 'vue'
import Vuex from 'vuex'
import { mount } from 'vue-test-utils'
import IsLoading from '@/components/Common/Notification/IsLoading'
import isLoadingModule from '@/store/modules/isLoading'

Vue.use(Vuex)

describe('Alert component rendering:', () => {
  let getters
  let store

  beforeEach(() => {
    getters = {
      isLoading: isLoadingModule.getters.isLoading
    }
  })

  it('Should render a hidden loading notification', () => {
    store = new Vuex.Store({
      getters,
      state: {
        isLoading: false
      }
    })

    const wrapper = mount(IsLoading, { store })
    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })

  it('Should render a visible loading notification', () => {
    store = new Vuex.Store({
      getters,
      state: {
        isLoading: true
      }
    })

    const wrapper = mount(IsLoading, { store })
    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })
})
