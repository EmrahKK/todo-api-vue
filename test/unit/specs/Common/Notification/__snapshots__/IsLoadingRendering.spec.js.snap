// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component rendering: Should render a hidden loading notification 1`] = `
<div class="loadingElement">
    Loading...
</div>
`;

exports[`Alert component rendering: Should render a visible loading notification 1`] = `
<div class="loadingElement loading">
    Loading...
</div>
`;
